vertx-mongo-client/src/test/java/io/vertx/ext/mongo/WatchTest.java

```diff
@@ -0,0 +1,49 @@
+package io.vertx.ext.mongo;
+
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.junit.Test;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class WatchTest extends MongoTestBase {
+
+  protected MongoClient mongoClient;
+
+  @Override
+  public void setUp() throws Exception {
+    super.setUp();
+    JsonObject config = getConfig();
+    mongoClient = MongoClient.createNonShared(vertx, config);
+    CountDownLatch latch = new CountDownLatch(1);
+    dropCollections(mongoClient, latch);
+    awaitLatch(latch);
+  }
+
+  @Override
+  public void tearDown() throws Exception {
+    mongoClient.close();
+    super.tearDown();
+  }
+
+  @Test
+  public void testInsertChangeStream() {
```

Comment:
Maybe slight name change.  We're asserting that our watch operation 'sees' the inset, correct?